### PR TITLE
Issue #2786 Amend SVG library to use XmlWriter instead of XmlTextWriter.

### PR DESCRIPTION
**Pull request by bbsgfalconer**
#### Reference Issue
Enhancement #693

#### What does this implement/fix? Explain your changes.
Generalize XML output to use an XmlWriter instead of the more specific XmlTextWriter. In particular, my use case is that I want to use my own custom XmlWriter to send output as part of an Asp.Net Blazor application.

The changes to the various WriteXXX(...) methods are trivial. However, the XmlWriter class performs some stricter checks than the XmlTextWriter and so I had to change the handling of namespaces. The existing code specifies namespace simply by writing an "xmlns" attribute. However, with XmlWriter, this causes an exception to be thrown (similar to [this question on StackOverflow](https://stackoverflow.com/questions/23698767/the-prefix-cannot-be-redefined-from-to-url-within-the-same-start-element-t) )

Note also the Microsoft documentation for [XmlTextWriter](https://docs.microsoft.com/en-us/dotnet/api/system.xml.xmltextwriter?view=netframework-4.8):

> Starting with the .NET Framework 2.0, we recommend that you use the XmlWriter class instead.

> Starting with the .NET Framework 2.0, we recommend that you create XmlWriter instances by using the XmlWriter.Create method and the XmlWriterSettings class to take advantage of new functionality.

See also conformance options based on [ConformanceLevel](https://docs.microsoft.com/en-us/dotnet/api/system.xml.xmlwritersettings.conformancelevel?view=netframework-4.8)

#### Any other comments?

The changes impacted SvgAttributeAttribute.cs such that:

* the SVG namespace (http://www.w3.org/2000/svg) is no longer included in the list of namespaces that would be written into the root SvgFragment element from this list. Instead, all SvgElements are explicitly using this namespace as per SvgElement.cs
* the 'NamespaceAndName' method is no longer needed. Instead, XmlWriter.WriteAttributeString is called with the namespace and prefix to handle this. (again, in SvgElement.cs)

Since the SvgNamespace is needed more widely across SvgElement and SvgAttributeAttribute, I extracted it into its own static class.

All unit tests pass.

### CODE REVIEW

**Comment by mrbean-bremen** [comment]
Looks good to me. @H1Gdev, @wieslawsoltes - can one of you have another look please?

**Comment by H1Gdev** [comment]
@mrbean-bremen 

I will check this PR.

File: Source/Document Structure/SvgFragment.cs [hunk 1]
```diff
@@ -298,16 +293,14 @@ public override SvgElement DeepCopy<T>()
         }
 
         // Override the default behavior, writing out the namespaces.
-        protected override void WriteStartElement(XmlTextWriter writer)
+        protected override void WriteStartElement(XmlWriter writer)
```
Comment (H1Gdev) [inline Source/Document Structure/SvgFragment.cs]: It also adds namespace to inner SVG.(existing issue...)
This process should be in `SvgDocument`.

File: Source/SvgElementFactory.cs [hunk 2]
```diff
@@ -113,7 +113,7 @@ private SvgElement CreateElement<T>(XmlReader reader, bool fragmentIsDocument, S
             else
             {
                 // All non svg element (html, ...)
-                createdElement = new NonSvgElement(elementName);
+                createdElement = new NonSvgElement(elementName, elementNS);
```
Comment (H1Gdev) [inline Source/SvgElementFactory.cs]: Does SVG embedded in HTML have an unnecessary namespace ?

**Comment by H1Gdev** [comment]
What is "Issue #2786" ?

File: Source/Document Structure/SvgFragment.cs (hunk 1 above)
Comment (H1Gdev) [reply Source/Document Structure/SvgFragment.cs]: Fix #702.

**Comment by H1Gdev** [comment]
Please separate PRs for each issue.

**Comment by bbsgfalconer** [comment]
Oops - tidying up my branches. Well separate into smaller changes and resubmit.

**Pull request closed without merging**